playbook/controllers.go

```diff
@@ -200,7 +201,7 @@ func HandlePlaybookList(c echo.Context) error {
 }
 
 func HandlePlaybookRunApproval(c echo.Context) error {
-	ctx := c.(api.Context)
+	ctx := c.Request().Context().(context.Context)
```

Comment:
Weren't we going to wrap the echo context ?

Reply:
We are wrapping it in the middleware: https://github.com/flanksource/incident-commander/blob/main/cmd/server.go#L77

Just need to type-assert in the funcs, or should I replace this with wrap as well ?

Reply:
```go
ctx := api.DefaultContext.Wrap(c.Request().Context())

// or

ctx := c.Request().Context().(context.Context)
```

Reply:
actually, we have to do it in the middleware itself cause the auth middleware needs duty.Context

Reply:
> We are wrapping it in the middleware: https://github.com/flanksource/incident-commander/blob/main/cmd/server.go#L77
> 
> Just need to type-assert in the funcs, or should I replace this with wrap as well ?

The way we are wrapping contexts in middleware is very confusing

`c.SetRequest(c.Request().WithContext(api.DefaultContext.Wrap(c.Request().Context())))`

Reply:
Should we create a func around it ?
```go
func WithEchoContext(c echo.Context, ctx dutycontext.Context) *http.Request {

    return c.Request().WithContext(
      ctx.Wrap(c.Request().Context())
  )
  ```
  
  And do  this in middleware ?
  ```go

c.SetRequest(WithEchoContext(c, api.DefaultContext))
```

Reply:
@yashmehrotra Not sure. We're still doing the same thing but in a different place. But also there doesn't seem to be a cleaner way to inject our context into echo's context.

We also could use `c.Set("duty-context", api.DefaultContext.Wrap(c.Request().Context())` although maybe that's not any better.

Reply:
> But also there doesn't seem to be a cleaner way to inject our context into echo's context.
yeah, i think lets leave it to original approach as its explicit